src/services/motion/types.ts

```diff
@@ -71,8 +71,9 @@ export const encodeMotionConfiguration = (
 ): pb.MotionConfiguration => {
   const result = new pb.MotionConfiguration();
 
+  const { obstacleDetectorsList = [] } = obj;
```

Comment:
How does this behave:
1.  if `obj` is `undefined`?
2. if `obstacleDetectorsList` is not a member of `obj`?

Reply:
1. The parameter type for `obj` for this function states that it will `never` be undefined. If that's not the case, it should be an optional param, and the param should recieve a default value:
```ts
export const encodeMotionConfiguration = (
  obj?: MotionConfiguration = {}
): pb.MotionConfiguration =>
```
2. If it's not a member of this object, it will be assigned a default value of `[]`.